mopidy_spotify/playlists.py

```diff
@@ -52,19 +56,36 @@ def _get_playlist(self, uri, as_items=False):
     def refresh(self):
         if not self._backend._web_client.logged_in:
             return
+        elif self._refreshing:
+            return
 
-        logger.info("Refreshing Spotify playlists")
+        self._refreshing = True
 
-        with utils.time_logger("playlists.refresh()", logging.DEBUG):
-            _sp_links.clear()
-            self._backend._web_client.clear_cache()
-            count = 0
-            for playlist_ref in self._get_flattened_playlist_refs():
-                self._get_playlist(playlist_ref.uri)
-                count = count + 1
-            logger.info(f"Refreshed {count} Spotify playlists")
+        logger.info("Refreshing Spotify playlists")
 
-        self._loaded = True
+        def refresher():
+            try:
+                with utils.time_logger("playlists.refresh()", logging.DEBUG):
+                    _sp_links.clear()
+                    self._backend._web_client.clear_cache()
+                    count = 0
+                    for playlist_ref in self._get_flattened_playlist_refs():
```

Comment:
@djmattyg007 How much have you been running this branch? I haven't validated this properly but looks to me like there may be concurrency issues here if a client attempts to enumerate the playlists in the middle of this operation (concurrent enumeration of a collection as it's modified).

Would it make sense to introduce a mutex here?